### DISCUSSION TITLE
I don't think I understand search 'results', why does it query the api again when I think it shouldn't need to?

### DISCUSSION THREAD

**Discussion by akda5id** [joalla/discogs_client#143] [Support] [answered]
I think I am doing something wrong, but I can't figure out what:
```
results = d.search(track=args.title, artist=args.artist, type='release')
	print('got results')
	page = results.page(1)
	if len(page) > 20:
		range_to_do = 20
	else:
		range_to_do = len(page)
	print('starting loop')
	for i in range(range_to_do):
		print(page[i].year)
```

It seems to me that year should be included in the search result per https://www.discogs.com/developers#page:database,header:database-search but it sure seems like the code above is hitting the api for each .year query, despite me saving the result page to a variable.

What am I missing?

**Comment by JOJ0** [comment, marked as answer]
There was some improvement on unnecesesary requests issues here https://github.com/joalla/discogs_client/issues/106 but I'm not sure if it's related to your issue. You might have hit a shortcoming of p3dc :-( 

Which version are you using btw?

**Reply by akda5id** [reply]
> There was some improvement on unnecesesary requests here #106

Ah, looking at that issue, does seem to be what I am hitting (year is unset for some of the records and DEBUG logging shows the requests coming then). A quick read of the fix for that issue looks like it attempted to fix only by looking if the request is the same as previous request, which wouldn't catch this, as it is a search (/database/search), vs. record (/releases/) request. Disappointing. The correct fix would be to know what fields are returned by search, and if they are not, to set them to `None`.

>Which version are you using btw?

discogs_client % head __init__.py 
__version__ = '2.7'

**Reply by JOJ0** [reply]
If you happen to be a Python programmer, we'd be happy if you could draft a PR with that idea and we try to work on it together. I'm currently limited with FOSS work resources but helping review such a PR within the next couple of weeks I could make.

**Reply by akda5id** [reply]
Not so much a programer, but my fix is up in #144 if you'd like to see. If that simple fix works for you, I could track down what else should be checked for and implement the same.
